files/build_templates/docker_image_ctl.j2

```diff
@@ -21,36 +21,26 @@ function getBootType()
     echo $TYPE
 }
 
-function postStartAction()
+function preStartAction()
 {
 {%- if docker_container_name == "database" %}
-    REDIS_SOCK="/var/run/redis/redis.sock"
-    until [[ $(/usr/bin/docker exec database redis-cli -s $REDIS_SOCK ping | grep -c PONG) -gt 0 ]]; do
-      sleep 1;
-    done
     WARM_DIR=/host/warmboot
-    SUDO="sudo -n"
     BOOT_TYPE=`getBootType`
     if [[ "$BOOT_TYPE" == "warm" && -d $WARM_DIR ]]; then
-        function redisLoadAndDelete()
-        {
-            DB="$1"
-            FILENAME="$2"
-            test -e $FILENAME || { echo "No file $FILENAME" >&2; exit 10; }
-            $SUDO redis-load -s $REDIS_SOCK -d $DB -e EMPTY $FILENAME || { echo "Failed to redis-load $FILENAME" >&2; exit 11; }
-            $SUDO rm $FILENAME || exit 12
-        }
-        # Load applDB from /host/warm-reboot/appl_db.json
-        redisLoadAndDelete 0 $WARM_DIR/appl_db.json
-        # Load configDB from /host/warm-reboot/config_db.json
-        redisLoadAndDelete 4 $WARM_DIR/config_db.json
-        # Load stateDB from /host/warm-reboot/state_db.json
-        redisLoadAndDelete 6 $WARM_DIR/state_db.json
-        # Load asicDB from /host/warm-reboot/asic_db.json
-        redisLoadAndDelete 1 $WARM_DIR/asic_db.json
-        # Load loglevelDB from /host/warm-reboot/loglevel_db.json
-        redisLoadAndDelete 3 $WARM_DIR/loglevel_db.json
+        # Load redis content from /host/warm-reboot/dump.rdb
+        test -e $WARM_DIR/dump.rdb && docker cp $WARM_DIR/dump.rdb database:/var/lib/redis/dump.rdb
```

Comment:
It seems the configdb-load.sh on database docker will load the config_db.json again, which is redundant to this dump.rdb, it may actually have different config if people forget to do cli save.   Should we do something special treatment for warm boot in in configdb-load.sh?

Reply:
Good catch! I fixed it.

---
In reply to: [235278619](https://github.com/Azure/sonic-buildimage/pull/2287#discussion_r235278619) [](ancestors = 235278619)